### PR TITLE
Make it possible to override Cache-Control

### PR DESCRIPTION
**Pull request by omnibs**
#### TLDR
* I want to override `Cache-Control` but can't, so I changed `default_headers` merge precedence to allow it

#### Reviewer asks
* Is it OK to keep the header override check in `rack_integration_test.rb`?
  - If not, how can I repro the non-override behavior in `rails_integration_test.rb`?

#### Details
Hi! I know it's advised not to cache the service-worker requests, but we have a case where we it's more desirable to prevent it from piercing our reverse-proxying CDN on Cloudflare than to have a guarantee the worker script is always up-to-date.

Trying to make that work I hit a problem with the way `opts` were merged into `default_headers`. `default_headers` took precedence and we could never override `Cache-Control`.

This PR proposes changing that.

I tried to include a test, initially in `rails_integration_test.rb`, but couldn't really reproduce the merge precedence problem, not sure why, maybe those tests don't go through the middleware initialization somehow?

I reproduced it successfully in `rack_integration_test.rb`, but from the looks of it I imagine it's intended as more of a smoke test setup then a place for validating features. Let me know if keeping it there is OK.

### CODE REVIEW

**Comment by omnibs** [comment]
Actually, this is not working for me :<

I'll open another PR when I get it right.

**Comment by omnibs** [comment]
Nvm, it does work, sorry for the confusion 😅

**Comment by rossta** [comment]
💜💜💜
